### ISSUE TITLE
Version 4.7.0 - Minor Release

### ISSUE THREAD

**Issue by MarshallOfSound** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3761]
This issue is for tracking progress on the release process.

- [x] Update MR_CHANGELOG
- [x] Version minor
- [x] Push to master
- [x] Build all platforms on CI
- [x] Check release process on staging update server `staging.googleplaymusicdesktopplayer.com`
- [x] Check operational on Windows
- [x]  Check operational on OS X
- [x] Check operational on Linux
- [x] Make release on GitHub
- [x] Upload to bintray
- [ ] ~Upload to WinStore~
- [ ] Watch the world burn :fire: :fire_engine:

**Comment by MarshallOfSound** [comment]
### RC Builds:

Please note that these builds are versioned as `4.7.0` and therefore won't auto update if the version we end up releasing is different to this RC.  Also the Windows installer is not code signed.

#### Darwin

* [ZIP File](https://3874-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip)


#### Linux

* [64 Deb](https://3874-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.7.0_amd64.deb)
* [32 Deb](https://3874-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.7.0_i386.deb)
* [64 RPM](https://3874-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.7.0.x86_64.rpm)
* [32 RPM](https://3874-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.7.0.i386.rpm)

#### Windows

* [Installer](https://ci.appveyor.com/api/buildjobs/y1jaaun4pn9xqwff/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)

**Comment by mlidbom** [comment]
I can verify that the Windows build works if that helps. :)

**Comment by galetahub** [comment]
Hi,
Ubuntu 19.10 (5.3.0-24-generic amd64) works fine :)
Thanks!

**Comment by trevorscottprice** [comment]
Verified the Windows build on my home PC. Nice work everyone.

**Comment by SoftWyer** [comment]
Chromecast button on Windows 10 is missing. Casting from other apps is OK on this machine.

Poked around with Bonjour and that seems to be OK. Anything I can try or see to identify the issue?

**Comment by Max9403** [comment]
Just to add onto what @SoftWyer said, my immediate thought was that it was because of using a Firefox UA but using the below UAs didn't show the cast option either, nor does it show in Opera, the closest I have to Chrome on hand (I didn't use a clean install to test though)

Chromium
>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/534.30 (KHTML, like Gecko) Ubuntu/11.04 Chromium/12.0.742.112 Chrome/12.0.742.112 Safari/534.30

Chrome
>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36

**Comment by MarshallOfSound** [comment]
I won't be able to debug Chromecast stuff till the second week of January.  Currently on holiday, barely got internet let alone a Chromecast.

If someone else could figure it out that'd be great, it's probably a client side UA check.  Using our preload script to hack the UA just for client side code should be enough (just guessing though)

**Comment by Max9403** [comment]
Never mind about my last comment it is indeed to do with the user agent since it hides the cast button if it can't find any devices (I forgot that I don't have one around at the moment).

It seems to just not be sending this script tag down when using the Firefox UA
><script type="text/javascript" src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1"></script>

which should be fixable with below snippet in PlayerPage in the _domReady function:
```javascript
document.querySelector("webview").executeJavaScript("let s = document.createElement('script'); s.src = 'https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1'; (document.head||document.documentElement).appendChild(s)");
```
Hope you're enjoying the holiday

**Comment by digitalcircuit** [comment]
4.7.0rc works on Kubuntu 19.10 64-bit!

Unfortunately, the media player information still appears to be broken, with media hotkeys not controlling playback and KDE Plasma's Media Player notification tray widget not showing up.  This *is* a minor issue and can be fixed next release.

I do get messages about the Avahi compatibility layer, though I think those have always been present:
```
*** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
```

If there's any further debugging I can do, I'd be happy to help.  I don't see any other messages, just `Starting Sentry` then the above repeated.

**Comment by MarkoShiva** [comment]
There is bug related to miniplayer. On linux if I try clicking on miniplayer with dark theme the window goes maximized and it start flickering cant get it back to normal except if I quit the player and then start again. 
Also yeah integration with player-control do not work so no playback control via multimediakeys.

**Comment by MarshallOfSound** [comment]
Thanks for testing folks, I've got new RC builds that _might_ fix the Chromecast thing but I took a shot in the dark.  It also might break the sign-in fix so if y'all could test the whole thing, signing in, using GPM, using YTM and trying to cast a thing that'd be really helpful.

### RC Builds:

Please note that these builds are versioned as `4.7.0` and therefore won't auto update if the version we end up releasing is different to this RC.  Also the Windows installer is not code signed.

#### Darwin

* [ZIP File](https://3883-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip)


#### Linux

* [64 Deb](https://3883-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.7.0_amd64.deb)
* [32 Deb](https://3883-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.7.0_i386.deb)
* [64 RPM](https://3883-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.7.0.x86_64.rpm)
* [32 RPM](https://3883-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.7.0.i386.rpm)

#### Windows

* [Installer](https://ci.appveyor.com/api/buildjobs/5ujn6oj83w6mkaek/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)

cc @MarkoShiva @digitalcircuit @Max9403 @trevorscottprice @galetahub @mlidbom @SoftWyer

**Comment by Jakobud** [comment]
This latest Windows build fixed the sign in issues in Win7 x64

**Comment by openmindead** [comment]
Works fine. Thanks!

**Comment by sorendam** [comment]
Great! This works on my Mac where it has not worked for months. Only thing is that it says that I signed in on Windows in a Firefox browser, but the IP is correct, so I guess that is just some sort of user agent hack to make this work. Great work!

**Comment by MarshallOfSound** [comment]
Yup that's just the user agent hack coming in to play.  I wonder if we should somehow let people know that that's a thing.  Or just assume folks will figure it out :/

**Comment by reduckted** [comment]
Probably worth adding something to the readme about it (and probably the changelog too). If nothing else, it will give you a single place to refer people to if you start getting questions about it.

**Comment by shaneh445** [comment]
Thank you so much for the fix. Never realize how much you love something until its gone. I should really start clicking the donate button once a year.

**Comment by SoftWyer** [comment]
I can confirm that the new Windows build has fixed the Chromecast issue, thanks for the quick turnaround!

**Comment by silentdis** [comment]
I'm happy to report functional on Kubuntu 19.10. Appears to be missing media access again, though... which is weird. I can access it just fine via KDEConnect on my smartphone, but from my widgets it just doesn't show up.

I'm uncertain which subsystem isn't seeing what's going on, here, as I'm not much of a hacker when it comes to the KDE backend stuff. Obviously *something* is working, or KDEConnect wouldn't work.

**Comment by subhead** [comment]
Windows 10 1909 Build 18363.535 works greaz again!

**Comment by Ralph77777** [comment]
4.7.0 GPM stuck loading music library, YTM loads fine. 
Windows 10 1909 18363.535

**Comment by solepixel** [comment]
MacOS download doesn't seem to be working for me. Getting a 500 Internal Server error at:
https://3865-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

**Comment by williamhscholl** [comment]
recently switched from spotify to GPM and this app is a lifesaver. was having trouble signing in like others but this version helped me sign in no problem.

thanks for your work on this!

**Comment by jarednthomas** [comment]
Ubuntu 19.04 checking in.

Removed all previous versions and 4.7.0 is logging in and working with no issues. 

Thanks for the fix!

**Comment by digitalcircuit** [comment]
[@silentdis](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3761#issuecomment-569728330 )
> I'm happy to report functional on Kubuntu 19.10. Appears to be missing media access again, though... which is weird. I can access it just fine via KDEConnect on my smartphone, but from my widgets it just doesn't show up.

After further experimentation, this matches my experience.  The KDE Plasma media player notification tray widget showed up the first time I tried on my desktop (separate from laptop), then it stopped ("No media playing") thereafter, but KDE Connect kept working.  Media keys still work, too, which they hadn't before on my laptop.

It's an improvement, at least :slightly_smiling_face:  I'll try to figure out what's consistently happening.

**Comment by b0rg0nd** [comment]
Windows 1809 (Build 17763.914) working

1. GPM Login working again
1. Casting to Google Home Mini working
1. Switch to YTM & hitting play > crashed the first time but consecutive switches worked fine, so _unable to reproduce_

👍

**Comment by DavidA122** [comment]
> MacOS download doesn't seem to be working for me. Getting a 500 Internal Server error at:
> https://3865-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

That link is working for me now if you want to try again, @solepixel 

I tried a couple of days ago and got the same 500 error, too.

**Comment by solepixel** [comment]
@DavidA122 Thanks! It's working now for me too!

**Comment by moraleida** [comment]
Works for me on Linux Mint 19.3 too, thank you!

**Comment by mikeyamato** [comment]
The Mac link, https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3761#issuecomment-569954006, does work. You'll also need to update settings in Security & Privacy to allow it to be [installed locally](https://support.apple.com/en-us/HT202491).

**Comment by byeuji** [comment]
> Thanks for testing folks, I've got new RC builds that _might_ fix the Chromecast thing but I took a shot in the dark. It also might break the sign-in fix so if y'all could test the whole thing, signing in, using GPM, using YTM and trying to cast a thing that'd be really helpful.
>
> #### Windows
> * [Installer](https://ci.appveyor.com/api/buildjobs/5ujn6oj83w6mkaek/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe)

Win10 x64 - Install failed initially, but succeeded when running the installer with as Admin. Login issues resolved.

**Comment by jostrander** [comment]
@byeuji This is probably due to it being unsigned.

**Comment by Zumochi** [comment]
The latest build from [here](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts) works for me on Windows 10 build 1909, thank you so much!

**Comment by dwhansen-cbg** [comment]
Works for me on Mac OS 10.15.2.

**Comment by dwhansen-cbg** [comment]
And thank you! Love this app.

**Comment by alitoyn** [comment]
Fixed my sign in issues on Linux 19.10 (64 deb). Thanks so much!

**Comment by Dragon911XX** [comment]
I installed to double check as I was having other issues with a different build (rameezv's build, issue #3640) not displaying the music title, artist and cover art (link below) on my new computer. Enable the Windows 10 System Media Service and Show track information in Windows 10 volume overlay are both checked.

![image](https://user-images.githubusercontent.com/7053735/71785757-0bbba400-2fc9-11ea-9279-f7ab92386509.png)

New minor release still is stuck loading library, so I'm unable to test if there's issues with volume overlay on 4.7.0.

**Comment by MarshallOfSound** [comment]
#### Mission Complete

Released as `4.7.1` --> https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/releases/tag/v4.7.1

**Comment by Silvenga** [comment]
🎉 

@MarshallOfSound are there still problems with the store release?
